code/game/atoms.dm

```diff
@@ -263,7 +262,33 @@ its easier to just keep the beam vertical.
 	if(desc)
 		user << desc
 
-	return distance == -1 || (get_dist(src, user) <= distance)
+	if(reagents)
+		if(reagent_flags & TRANSPARENT)
+			to_chat(user, "<span class='notice'>It contains:</span>")
+			if(reagents.reagent_list.len)
+				for(var/I in reagents.reagent_list)
+					var/datum/reagent/R = I
+					to_chat(user, "<span class='notice'>[R.volume] units of [R.name]</span>")
+
+				// TODO: reagent vision googles? code below:
```

Comment:
I hope we don't forget about this TODO because reagent vision goggles are very useful.

Reply:
Right now, anyone can just see reagents regardless.